essay/swar/latency.md

```diff
@@ -107,24 +107,48 @@ Curiously enough, our SWAR library gives that capability directly; in the `strle
+
+Just like I took a comment that reflects the modesty of Professor Lemire as inspiration to discuss at lenght the relation between logarithmic latencies and parallelism, I think I should criticize dear Knuth's title of calling this "tricks":
+
+His "broadword" (what we call SWAR) remarks remind me of the famous essay by Nobel Price Winner in Physics, 1963, Eugene Wigner's famous essay "The Unreasonable Effectiveness of Mathematics in the Natural Sciences", it says something like
+
+> The miracle of the appropriateness of the language of mathematics for the formulation of the laws of physics is a wonderful gift which we neither understand nor deserve
+
+Which seems the same idea, the wonder that it works.  I prefer to side with Richard P. Feynman, also NPW, and perhaps more famous than E. Wigner, that for him, mathematics is just "organized reasoning".  Then Wigner's point of view gets reduced by Feynman to the apparent fact that nature is amenable to be described by reasoning, or that nature is reasonable.  This begs for a Wittgenstein's retort of how would an unreasonable nature would look like given that all we know is our apparently reasonable universe --complex numbers end up making sense after deep study, the sooner you reject the perspective that real numbers describe the universe the sooner you'd regain your senses, the universe is not unreasonable, it is that it is complex numbers what describe the universe, not the too-simple real numbers--
+
+Fortunately, as software engineers, we know what it looks like when SIMD is not approached from a principled stand point, you get Intel's SSE, AVX, AVX512: an unholy mess of marketing-driven engineering (Google for what Agner Fog says about this design, and all of Intel's ISAs for that matter); back to SWAR, it really doesn't surprise me the effectiveness of integer arithmetic in combination with bitwise operations to express tactical parallelism: they are what was designed and proved to work as the basis for all computation.  In the end, it is all functions with binary values, binary functions are the simplest, **where you have to be smart is into combining them in ways that are combinable further**
```

Comment:
s/stand point/standpoint
s/where you have to be smart is into combining them in ways that are combinable further/where you have to be smart is in combining them in ways that are combinable further/
or maybe 'combining them in ways that allow more combinations' or something.